src/main/java/com/sublinks/sublinksapi/person/services/PasswordResetService.java

```diff
@@ -0,0 +1,65 @@
+package com.sublinks.sublinksapi.person.services;
+
+import com.sublinks.sublinksapi.person.dto.PasswordReset;
+import com.sublinks.sublinksapi.person.dto.Person;
+import com.sublinks.sublinksapi.person.events.PasswordResetCreatedEventPublisher;
+import com.sublinks.sublinksapi.person.events.PasswordResetUpdatedEventPublisher;
+import com.sublinks.sublinksapi.person.repositories.PasswordResetRepository;
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class PasswordResetService {
+
+  private final PasswordResetRepository passwordResetRepository;
+  private final PasswordResetCreatedEventPublisher passwordResetCreatedEventPublisher;
+  private final PasswordResetUpdatedEventPublisher passwordResetUpdatedEventPublisher;
+
+  public Optional<PasswordReset> findFirstByToken(String token) {
+
+    return passwordResetRepository.findFirstByToken(token);
+  }
+
+  public String generateToken() {
+
+    return UUID.randomUUID().toString();
+  }
+
+  public PasswordReset createPasswordReset(Person person) {
+
+    PasswordReset passwordReset = new PasswordReset();
+    passwordReset.setPerson(person);
+    passwordReset.setUsed(false);
+    passwordReset.setCreatedAt(new Date());
+    passwordReset.setToken(generateToken());
```

Comment:
This is a _bit_ of a nit, but I would recommend against using UUIDs for a password reset token.

The Java UUID generator does make an effort to prevent predictability but that not its primary objective: https://docs.oracle.com/javase/8/docs/api/java/util/UUID.html#randomUUID

A better approach would likely be to generated N number of bytes from SecureRandom

https://stackoverflow.com/a/56705603

Another strategy here would be cryptographically signing the generated ID (i.e., outright issue a scoped JWT)

https://github.com/sublinks/sublinks-api/blob/6afdd061aa698a9bb7a7412a0b5535511c746f0c/src/main/java/com/sublinks/sublinksapi/api/lemmy/v3/authentication/JwtUtil.java#L71

That would require that both the ID is predicted _and_ signed with the server's private key which _should_ be next to impossible to perform without a broader system compromise.